### DISCUSSION TITLE
Could not poll the database: Base table or view not found

### DISCUSSION THREAD

**Discussion by terrencetingleberry** [firefly-iii/firefly-iii#8306] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

I'm unsure if this is an issue with Firefly III or if I've somehow made a mistake while configuring it.

### Description

I configured Firefly III along with data-importer using Docker for the first time using [this tutorial](https://docs.firefly-iii.org/how-to/data-importer/installation/docker/) from the official documentation and ran into the following error when trying to open the web portal:

Could not poll the database: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (Connection: mysql, SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)

### Debug information

Unfortunately, I'm unable to gather any logs as docker says the container doesn't exist whenever I attempt to fetch the logs, even though it shows the container when I run `docker ps` in terminal.

### Expected behaviour

The standard, blank web portal appears with no error page.

### Steps to reproduce

1. Install Firefly + data-importer using [this tutorial](https://docs.firefly-iii.org/how-to/data-importer/installation/docker/).
2. Open _localhost_ in Firefox.

### Additional info

I'm running Pop!_OS 22.04 LTS (Ubuntu 22.04 LTS) and am extremely new to Docker and local-hosting as a whole.

**Comment by JC5** [comment]
What's the output of `docker container ls -a`? That should show you all containers, even the stopped ones. You can show the logs of any container by doing `docker logs [containerid]`, which you can get from the previous command.

**Comment by terrencetingleberry** [comment]
> What's the output of `docker container ls -a`?

[containers.txt](https://github.com/firefly-iii/firefly-iii/files/13797343/containers.txt)

> You can show the logs of any container by doing docker logs [containerid]

[log.txt](https://github.com/firefly-iii/firefly-iii/files/13797337/log.txt)

**Comment by JC5** [comment]
Alright so you keep getting `Connection refused` or table not found errors. What's in the logs of the database container?

**Reply by terrencetingleberry** [reply]
[db_log.txt](https://github.com/firefly-iii/firefly-iii/files/13797362/db_log.txt)

**Comment by JC5** [comment]
Did you ever change the database password in any of the .env files?

**Reply by terrencetingleberry** [reply]
Yes, I did.

**Reply by terrencetingleberry** [reply]
I'm going to try going through [the standard setup process](url) for the time being until the issue with the Docker install is solved. Running it through a Docker container would be much more preferable long-term, but hopefully a standard installation will suffice for now and not give me an error.

**Reply by terrencetingleberry** [reply, marked as answer]
Nevermind, I deleted the container and images and built them again and the error is gone. I went to the web portal and was greeted with a login page, I'm not sure what went wrong before. I changed nothing about the files and simply ran the build command again, odd. Anyway, thanks for the help! This can be considered solved.

**Comment by giuseppebaldi** [comment]
I am having this same exact error and cannot figure out what might be causing it. I have deleted the container/images and built them again several times but keep getting this same error: Could not poll the database: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (Connection: mysql, SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1). I have also tried to restart the container a few times to no avail.

I also followed the same tutorial so I wonder if there's an issue there somewhere.

Database container log:
[firefly_iii_db logs.txt](https://github.com/user-attachments/files/17689954/firefly_iii_db.logs.txt)

Output of docker container ls -a
[containers.txt](https://github.com/user-attachments/files/17689960/containers.txt)

Any idea what might cause this issue? TIA

**Reply by giuseppebaldi** [reply]
I was able to fix this by restarting the firefly_iii-core container and allowing it to go through the migration process, which took a while but I just watched the logs until it was done. Just in case someone else has the same issue later.

**Reply by gawkward** [reply]
> I was able to fix this by restarting the firefly_iii-core container and allowing it to go through the migration process, which took a while but I just watched the logs until it was done. Just in case someone else has the same issue later.

My God I wish I read to the bottom of this thread before trying a million other things. It would have saved so much time... I can confirm that simply restarting firefly_iii_core resolved this issue for me too. Thank you!

**Comment by Fabeson** [comment]
For anyone else running into "Could not poll the database: SQLSTATE[42S02]: Base table or view not found" and neither restarting nor re-deploying/deleting containers helps (as was my case):

You can enter the Firefly core container itself:
`docker exec -it firefly_iii_core bash`

and then trigger the migration process manually:

`#Run migrations (this will create all tables)`
`php artisan migrate --force`

`#Seed the database with initial data`
`php artisan db:seed --class=DatabaseSeeder`
